src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -11,14 +21,72 @@
 
 public class ActivityFeedServlet extends HttpServlet {
 	
+	  /** Store class that gives access to Users. */
+  private UserStore userStore;
+
+	 /** Store class that gives access to Messages. */
+  private MessageStore messageStore;
+
+  /** Store class that gives access to Conversations. */
+  private ConversationStore conversationStore;
+
+  /**
+   * Set up state for handling conversation-related requests. This method is only called when
+   * running in a server, not when running in a test.
+   */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+		setMessageStore(MessageStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+	/**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }	
+	
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method
+   * for use by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+	
 	/**
 	*This function fires when a user requests the /activityfeed URL. It forwards the
 	*request to activityfeed.jsp.
 	*/
 	@Override
 	public void doGet(HttpServletRequest request, HttpServletResponse response) 
 			throws IOException, ServletException {
+		
+		Conversation conversation = conversationStore.getActFeedConversation();
+   
+    UUID conversationId = conversation.getId();
+
```

Comment:
Is getting the conversationId used later for checking if the conversation is valid // do we need this right now for doGet()?

Reply:
It gets the activity feed's conversation id, which allows us to retrieve the list of messages we want a little later on (i'm retrieving the messages through their conversationId attribute, which I think is the only way they'll let us do it)

Reply:
oh you're right, I'm so sorry I didn't notice the next line >< looks good!